lib/spack/spack/test/concretization/core.py

```diff
@@ -3257,6 +3284,36 @@ def test_spec_unification(unify, mutable_config, mock_packages):
         _ = spack.cmd.parse_specs([a_restricted, b], concretize=True)
 
 
+@pytest.mark.usefixtures("mutable_config", "mock_packages", "do_not_check_runtimes_on_reuse")
+@pytest.mark.parametrize(
+    "spec_str, should_pass, error_type",
+    [
+        # TODO write actual Exceptions for these to give good error messages
+        (f"git-ref-package@main commit={'a' * 40}", True, None),
+        (f"git-ref-package@main commit={'a' * 39}", False, AssertionError),
+        (f"git-ref-package@2.1.6 commit={'a' * 40}", False, AssertionError),
+        (f"git-ref-package@git.2.1.6=2.1.6 commit={'a' * 40}", True, None),
```

Comment:
```suggestion
        (f"git-ref-package@git.2.1.6=2.1.6 commit={'a' * 40}", True, None),
        (f"git-ref-package@git.2.1.6 commit={'a' * 40}", True, None),        
```
I think we should add one additional case here.

Reply:
Ugh, when should we deprecate the ref lookup?

Reply:
@becker33  I have to re-write this whole test to use one of the git fixtures that locally generate a git history so it can do the lookup. 

Considering we are 

1. planning to deprecate/remove ref lookup and 
2. actively discourage users from using it

 I'll push back on this refactor a bit. Do we need this test? If so I will do it, but I would rather not.